packages/node/src/node/NodePhysicalProperties.ts

```diff
@@ -0,0 +1,90 @@
+import { PowerSource } from "#clusters/power-source";
+import { ThreadNetworkDiagnostics } from "#clusters/thread-network-diagnostics";
+import { Endpoint } from "#endpoint/index.js";
+import { DeviceClassification, IcdManagement } from "#model";
+import { Node } from "#node/Node.js";
+import { PhysicalDeviceProperties } from "#protocol";
+import { ClusterId } from "#types";
+
+/**
+ * Inspects a node to generate {@link PhysicalDeviceProperties}.
+ */
+export function NodePhysicalProperties(node: Node) {
+    const rootEndpointServerList = [...node.state.descriptor.serverList];
+
+    const properties: PhysicalDeviceProperties = {
+        threadConnected: false,
+        wifiConnected: false,
+        ethernetConnected: false,
+        rootEndpointServerList,
+        isBatteryPowered: false,
+        isIntermittentlyConnected: rootEndpointServerList.includes(IcdManagement.id as ClusterId),
+        isThreadSleepyEndDevice: false,
+    };
+
+    inspectEndpoint(node, properties);
+
+    return properties;
+}
+
+function inspectEndpoint(endpoint: Endpoint, properties: PhysicalDeviceProperties) {
+    // Network interface support
+    const network = endpoint.behaviors.typeFor(NetworkCommissioningClient);
+    if (network) {
+        const features = network.schema.supportedFeatures;
+        if (features.has("WI")) {
+            properties.wifiConnected = true;
+        }
+        if (features.has("TH")) {
+            properties.threadConnected = true;
+        }
+        if (features.has("ET")) {
+            properties.ethernetConnected = true;
+        }
+    }
+
+    // Battery power
+    const powerSource = endpoint.behaviors.typeFor(PowerSourceClient);
+    if (powerSource) {
+        const features = powerSource.schema.supportedFeatures;
+        if (
+            features.has("BAT") ||
+            // Perform additional checks because we've encountered devices with incorrect features
+            !features.has("WIRED") ||
+            endpoint.behaviors.elementsOf(powerSource).attributes.has("batChargeLevel")
+        ) {
+            if (endpoint.stateOf(PowerSourceClient).status === PowerSource.PowerSourceStatus.Active) {
+                properties.isBatteryPowered = true;
+            }
+        }
+    }
+
+    // Sleepy thread device
+    const threadNetworkDiagnostics = endpoint.behaviors.typeFor(ThreadNetworkDiagnosticsClient);
+    if (
+        threadNetworkDiagnostics &&
+        endpoint.stateOf(threadNetworkDiagnostics).routingRole === ThreadNetworkDiagnostics.RoutingRole.SleepyEndDevice
+    ) {
+        properties.isThreadSleepyEndDevice = true;
+    }
+
+    // Recurse into children
+    //
+    // Only consider the root endpoint and utility endpoints.  Clusters on application endpoints or sub-endpoints refer
+    // to a different device
+    for (const part of endpoint.parts) {
+        if (part.number !== 0 && part.type.deviceClass !== DeviceClassification.Utility) {
```

Comment:
Are multiple device types aggregated here to include the proper details? i did not checked the relevant code there now

Reply:
They're not aggregated, no, but I was assuming this is the primary device type, which should always be the most specific application type on application endpoints.  If that's not the case then we can fix that

Reply:
Difficult, especially when you think about power source. in fact you can place a power source on root or on an extra endpoint or on the application endpoint, but then it would secondary anyway.
And TBH when we redesigning now, maybe lets ignore "bridged nodes" at all because they might have stuff that confuses us :-) (like "battery info for a sub thing)

Reply:
Wait the issue I was referring to is the order of `deviceTypeList`.  If a utility device type was for some reason listed first then we wouldn't notice the application type.  I adjusted in separate PR to ensure we prefer the application device types, but that also turned up that I wasn't actually setting the device classification for ClientNode so fixed that too.

The point of this code is to ignore application endpoints and all of their children for these purposes.  So that skips aggregators and thus bridged nodes.

The comment "Cluster from application endpoint should not affect default" in `NodePhysicalProperties` is where this is tested, although it's running against `ServerNode` so missed the fact that it wouldn't have worked on `ClientNode`.

Reply:
But as said power source might be on an application endpoint. Then we would. It see it potentially